src/ejabberd_c2s.erl

```diff
@@ -3104,8 +3124,7 @@ maybe_add_timestamp({F, T, #xmlel{name= <<"message">>}=Packet}=PacketTuple, Time
         <<"headline">> ->
             PacketTuple;
         _ ->
-            %% TODO: ?MYNAME (or server taken from c2s state) not <<"localhost">>
```

Comment:
TODO should remain here, as actually the right domain to use is the server from c2s state.

Reply:
just use SD#state.server here (pass Server as an extra argument)